### PR TITLE
Fixing small issues with tagged hashes

### PR DESCRIPTION
**Pull request by dr-orlovsky**
The branch failed to compile o the nightly rust, so I have figured out what have to be fixed. Also made namespaces more consistant for trait "inheritance"

### CODE REVIEW

File: src/sha256t.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ use Hash as HashTrait;
 use Error;
 
 /// Trait representing a tag that can be used as a context for SHA256t hashes.
-pub trait Tag: Copy + Ord + Default + ::core::hash::Hash {
+pub trait Tag: Copy + Ord + Default + HashTrait {
```
Comment (stevenroose) [inline src/sha256t.rs:25]: This is not correct. `Tag` is not a hash...

File: src/sha256t.rs [hunk 2]
```diff
@@ -42,8 +42,8 @@ impl<T: Tag> HashTrait for Hash<T> {
     type Engine = sha256::HashEngine;
     type Inner = [u8; 32];
 
-    fn engine() -> sha256::HashEngine {
```
Comment (stevenroose) [inline src/sha256t.rs:45]: Since the hash type is `sha256t`, it's for tagged sha256 hashes. So this can stay as the engine.

**Review by stevenroose — CHANGES_REQUESTED** [review]
I don't think this is useful. The issue currently with tagged hashes is the serde function.

File: src/sha256t.rs (hunk 1 above)
Comment (dr-orlovsky) [reply src/sha256t.rs:25]: I had a compilation issues before I changed this, will re-check what it was. In my project I'm using tagged hashes the same way I use other hashes, and some macro, generic by a hash type, was failing due to this line (of Tag being not Hash, but being a part of `sha256t::Hash`). Will double-check that and follow up.

Comment (dr-orlovsky) [reply src/sha256t.rs:25]: Ok, I messed up; the problem was caused by another factor. Closing the request

**Pull request closed without merging**